spec/models/tag_spec.rb

```diff
@@ -232,5 +232,23 @@
 
       expect(results).to eq [tag, similar_tag]
     end
+
+    it 'finds only listable tags' do
+      tag = Fabricate(:tag, name: 'match')
+      _miss_tag = Fabricate(:tag, name: 'matchunlisted', listable: false)
+
+      results = described_class.search_for('match')
+
+      expect(results).to eq [tag]
+    end
+
+    it 'finds non-listable tags as well via option' do
+      tag = Fabricate(:tag, name: 'match')
+      unlisted_tag = Fabricate(:tag, name: 'matchunlisted', listable: false)
+
+      results = described_class.search_for('match', 5, 0, include_unlistable: true)
```

Comment:
Arguably the limit / offset here should probably be via an option as well, but that involves modifying code well out of scope of this PR, hence leaving as is for now.